ethereum_history_api/circuits/lib/src/rlp/types_test.nr

```diff
@@ -49,40 +76,160 @@ mod rlp_fragment {
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP type")]
+        fn invalid_type() {
+            let invalid_type_fragment = RlpFragment { offset: 0, length: 0, data_type: LIST };
+
+            invalid_type_fragment.assert_eq_null("Field")
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length() {
+            let invalid_length_fragment = RlpFragment { offset: 0, length: 1, data_type: STRING };
+
+            invalid_length_fragment.assert_eq_null("Field")
+        }
+    }
+
+    mod assert_eq_u1 {
         use crate::rlp::{decode::STRING, types::RlpFragment};
-        use crate::misc::types::Bytes32;
-        use crate::misc::arrays::alter_array;
 
-        global bytes = [
-            0x88, 0xe9, 0x6d, 0x45, 0x37, 0xbe, 0xa4, 0xd9, 0xc0, 0x5d, 0x12, 0x54, 0x99, 0x07, 0xb3, 0x25, 0x61, 0xd3, 0xbf, 0x31, 0xf4, 0x5a, 0xae, 0x73, 0x4c, 0xdc, 0x11, 0x9f, 0x13, 0x40, 0x6c, 0xb6
-        ];
+        global byte = 1;
+        global rlp = [0x01];
+        global fragment = RlpFragment { offset: 0, length: 1, data_type: STRING };
 
-        global fragment = RlpFragment { offset: 0, length: 32, data_type: STRING };
+        #[test]
+        fn succes() {
+            fragment.assert_eq_u1("Field", rlp, byte)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length() {
+            let rlp = [0x01, 0x01];
+            let invalid_length_fragment = RlpFragment { offset: 0, length: 2, data_type: STRING };
+
+            invalid_length_fragment.assert_eq_u1("Field", rlp, byte)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP value")]
+        fn invalid_value() {
+            fragment.assert_eq_u1("Field", rlp, 0)
+        }
+    }
+
+    mod assert_eq_u8 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0x11];
+        global fragment = RlpFragment { offset: 0, length: 1, data_type: STRING };
+        global value = 17;
 
         #[test]
         fn success() {
-            fragment.assert_eq_bytes32("Field", bytes, bytes)
+            fragment.assert_eq_u8("Field", rlp, value)
         }
 
         #[test(should_fail_with = "Field: Invalid RLP length")]
         fn invalid_length() {
+            let rlp = [0x01, 0x01];
             let invalid_length_fragment = RlpFragment { offset: 0, length: 2, data_type: STRING };
 
-            invalid_length_fragment.assert_eq_bytes32("Field", bytes, bytes)
+            invalid_length_fragment.assert_eq_u8("Field", rlp, value)
         }
 
         #[test(should_fail_with = "Field: Invalid RLP value")]
         fn invalid_value() {
-            fragment.assert_eq_bytes32("Field", bytes, alter_array(bytes))
+            fragment.assert_eq_u8("Field", rlp, 16)
+        }
+    }
+
+    mod assert_eq_u32 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0xEE, 0x6B, 0x28, 0x00]; //4000000000
```

Comment:
Also - this is not RLP. This is value as u8. encoded_value?

Reply:
As you see - it doesn't have RLP prefix

Reply:
You're right. But shouldn't we then use real rlp in tests? After all we are testing RlpFragment and in all of the assert_eq_t functions this argument is called rlp, because it expects rlp. What do you think?

Reply:
RlpFragment is a fragment of data decoded from RLP. It does not have Rlp left in it. Ideally - we would use normal fragment instead of Rlp fragment and we tried, but it's not performant. So rlp fragment is essentially a fat pointer (pointer with length).